llvm/lib/Target/BPF/BPFInstrInfo.td

```diff
@@ -604,6 +604,32 @@ class BRANCH_LONG<BPFJumpOp Opc, string OpcodeStr, list<dag> Pattern>
   let BPFClass = BPF_JMP32;
 }
 
+class BRANCH_OR_NOP<BPFJumpOp Opc, string OpcodeStr, list<dag> Pattern>
```

Comment:
these two insns are just special cases for `JMPL`, how about just extending(inherit) `BRANCH_LONG` and override `Inst{55-52}` fields?

Reply:
> these two insns are just special cases for `JMPL`, how about just extending(inherit) `BRANCH_LONG` and override `Inst{55-52}` fields?

Thanks. @inclyc Good suggestion and adding the test! I marked the patch as RFC. Will wait until the design in kernel is settled and then will make proper coding (as you suggested in the above) and add tests.